src/org/wikipedia/Wiki.java

```diff
@@ -575,9 +617,15 @@ public static Wiki newSession(String domain, String scriptPath, String protocol)
      */
     protected void initVars()
     {
-        base = protocol + domain + scriptPath + "/index.php";
-        apiUrl = protocol + domain + scriptPath + "/api.php";
-        articleUrl = protocol + domain + "/wiki/";
+        try {
+            base = new URI(protocol + domain + scriptPath + "/index.php").normalize().toString();
+            apiUrl = new URI(protocol + domain + scriptPath + "/api.php").normalize().toString();
+            articleUrl = new URI(protocol + domain + articlePath).normalize().toString();
+        } catch(URISyntaxException e) {
+            // Log and exit
+            log(Level.SEVERE, "initVars", "Provided URI is invalid: " + e.getMessage() + "\nProgram exiting.");
+            System.exit(1);
```

Comment:
Throwing an `IllegalArgumentException` would be a better option. This library is used in several graphical front ends where exiting is not a desirable behavior.